samples/OpenIdConnectSample/Program.cs

```diff
@@ -1,24 +1,46 @@
 ﻿using System.IO;
+using System.Reflection;
+using System.Security.Cryptography.X509Certificates;
 using Microsoft.AspNetCore.Hosting;
-using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.FileProviders;
 
 namespace OpenIdConnectSample
 {
     public static class Program
     {
         public static void Main(string[] args)
         {
-            var config = new ConfigurationBuilder().AddEnvironmentVariables("ASPNETCORE_").Build();
-
             var host = new WebHostBuilder()
-                .UseKestrel()
-                .UseConfiguration(config)
+                .UseKestrel(options =>
+                {
+                    //Configure SSL
+                    var serverCertificate = LoadCertificate();
+                    options.UseHttps(serverCertificate);
+                })
                 .UseContentRoot(Directory.GetCurrentDirectory())
                 .UseIISIntegration()
                 .UseStartup<Startup>()
                 .Build();
 
             host.Run();
         }
+
+        private static X509Certificate2 LoadCertificate()
+        {
+            var assembly = typeof(Startup).GetTypeInfo().Assembly;
+            var embeddedFileProvider = new EmbeddedFileProvider(assembly, "OpenIdConnectSample");
```

Comment:
nit: why using `EFP` instead of directly loading the certificate from the assembly?